common/downstream_fatal_error.md

```diff
@@ -0,0 +1,131 @@
+> _pyRACF expects IRRSMO00 to return a string containing XML data. If IRRSMO00 returns an empty string, or if the **SAF Return Code** in the result XML is greater than `4`, `DownstreamFatalError` is raised._
+
+&nbsp;
+
+pyRACF expects IRRSMO00 to return a non-empty response string after processing a request. If IRRSMO00 returns an empty response string or the **SAF Return Code** is greater than `4`, indicating that there was an issue with command image processing, a `DownstramFatalError` will be raised. It is possible that RACF commands are executed in this situation, but this is not the case for any known causes of this error.
+
+## RACF Authorizations
+
+&nbsp;
+
+A common cause of this error is that the user does not have the proper RACF authorizations as outlined in [our dependencies note](../../index). One of the possible failures is the user not having `READ` authority to the `IRR.IRRSMO00.PRECHECK` resource in the `XFACILIT` class, which is required for `set` or `alter` operations. If you are not certain if you have this authority or if this has been established in your environment, please consult our [Check for & set up RACF Authorizations](../setup_precheck) documentation. Another possibility is the user not having `ALTER` authority to the `userid.IRRSMO00` resource in the `SURROGAT` class, which is required to run pyRACF commands as another userid. Please review our [run as userid](../run_as_userid) documentation for more information on this feature.
+
+###### Python Script
+```python
+from pyracf import UserAdmin
+from pyracf import DownstreamFatalError
+
+user_admin = UserAdmin()
+
+try:
+    user_admin.set_uid("squidwrd", 123456)
+except DownstreamFatalError as e:
+    print(e.message)
+```
+
+###### Console Output
+```console
+Security request made to IRRSMO00 failed.
+
+SAF Return Code: 8 
+RACF Return Code: 200 
+RACF Reason Code: 16
+
+Check to see if the proper RACF permissions are in place.
+For `set` or `alter` functions, you must have at least READ access to `IRR.IRRSMO00.PRECHECK` in the `XFACILIT` class.
+```
+
+###### Python Script
+```python
+from pyracf import UserAdmin
+from pyracf import DownstreamFatalError
+
+user_admin = UserAdmin(run_as_userid="ESWIFT")
+
+try:
+    user_admin.set_uid("squidwrd", 123456)
+except DownstreamFatalError as e:
+    print(e.message)
+```
+
+###### Console Output
+```console
+Security request made to IRRSMO00 failed.
+
+SAF Return Code: 8 
+RACF Return Code: 200 
+RACF Reason Code: 8
+
+Check to see if the proper RACF permissions are in place.
+For the `run_as_userid` feature, you must have at least UPDATE access to `ESWIFT.IRRSMO00` in the `SURROGAT` class.
+```
+
+## Improper Use of Traits
+
+&nbsp;
+
+As stated, this error can also appear if there is a problem with the data pyRACF passes to IRRSMO00. In this situation, you can review the `request_xml` and the `result_dictionary`, which are stored as attributes in the `DownstreamFatalError` exception object for more information.

```

Comment:
```markdown
As stated, may also be raised if there is a problem with the data pyRACF passes to IRRSMO00. In this situation, you can review `DownstreamFatalError.request_xml` and the `DownwstreamFatalError.result_dictionary` for more information.
```

Reply:
sure. This is clearer.

Reply:
My bad. This should be 
```markdown
As stated, `DownstreamFatalError` may also be raised
```
not 
```markdown
As stated, may also be raised
```

Reply:
sorry, should have caught that. Changed.